src/Logary.CSharp.Facade/Facade.cs

```diff
@@ -471,15 +471,16 @@ public Flyweight(string[] name)
 
             T WithLogger<T>(Func<ILogger, T> callback)
             {
-                var local = _config;
-                var fwCurr = _fwClock;
-
-                if (fwCurr != local.Item2)
+                if (_fwClock != _config.Item2)
                 {
                     lock (_updating)
                     {
-                        _logger = local.Item1.GetLogger(_name);
-                        _fwClock += 1u;
+                        var local = _config;
+                        if(_fwClock != local.Item2)
+                        {
+                            _logger = local.Item1.GetLogger(_name);
+                            _fwClock = local.Item2;
```

Comment:
every time we initialize config, the clock increases by 1. and when we log message, we will check if the local config clock same as the current config's clock , if not, get new logger from new config and update the clock then log message use the new logger .

e.g.
1. the default fwClock is 1.
1. initialize logging config ,  config clock will be 2
1. if we initialize logging config twice here or between some time,  config clock will be 3
1. then log message through facade api, it should be updating config and clock once. 

so assign the current clock to the fwClock (not by adding one) can avoid more checking .